### PR TITLE
Add coveralls

### LINKED ISSUE #70: Add coveralls
Add coverrals for test coverage

### LINKED ISSUE #204: Add coveralls badge on README
Add coveralls badge on README so report of the test coverage is displayed
## Expected Behavior

A badge is added to README to that displays the test coverage percentage report
## Current Behavior

There is no badge that displays test coverage
## Possible Solution

Use the code below maybe?
`[Coverage Status](https://coveralls.io/github/AgileVentures/sfn-client?branch=70_add-coverrals)`
[This Link](https://medium.com/the-code-review/add-badges-for-travisci-coveralls-and-code-climate-to-your-readme-ruby-6d27caad74ed) may guide
## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39):
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by aonomike**
Fixes #70 
Fixes #204

### CODE REVIEW

File: .coveralls.yml
Comment (yakryder) [inline .coveralls.yml]: A few questions

1) Is this file necessary?
2) Is this sensitive information that should be excluded from source?
3) If the answer to question two is yes, how would you recommend proceeding?

File: jest.config.js
Comment (yakryder) [inline jest.config.js]: Seems like cypress is in here twice

File: jest.config.js
Comment (yakryder) [inline jest.config.js]: See above.

Do you see a reason for having cypress both in the 'don't run tests' and 'don't generate coverage' sections?

File: package.json
Comment (yakryder) [inline package.json:31]: I would recommend this be adjusted

Personally I would hope to see these new coveralls scripts namespaced with `test`. 

Barring that, I would at least hope that there weren't testing-specific scripts both namespaced under `test` (like `test:debug`) and not namespaced under `test` (like added scripts `coverage` and `coveralls`).

See note below for a possible motivation for removing the `coverage` script

File: jest.config.js
Comment (yakryder) [inline jest.config.js]: This appears to be deprecated. Please find evidence to the contrary or an alternative implementation

File: package.json
Comment (yakryder) [inline package.json]: It seems like you have already setup the jest config to default to collecting coverage. If that is so, I don't see the point in adding a script for this.

File: jest.config.js
Comment (yakryder) [inline jest.config.js]: This defaults to false, so you seem to be changing the behavior of the test suite here in a way not required to do the work. I'd prefer not to see something like that rolled in with other work without an accompanying justification.

File: .travis.yml
Comment (yakryder) [inline .travis.yml:13]: Should already default to collecting coverage via the jest config, however an argument can be made here for explicitness.

File: package.json
Comment (yakryder) [inline package.json]: I might prefer to see this as `test:report_coverage` or something of the sort. The idea being to both make clearer what this script is doing and not unnecessarily couple our interface to that of our coverage tracking tool.

There are definitely counterarguments to be made.

**Review by yakryder — COMMENTED** [review]
@aonomike Thanks for your work and patience.

Biggest potential issue I see is the `.coveralls.yml`. Both my own intuition and a cursory Google suggested to me that this is likely not a great way to do this. The file itself may not even be necessary. 

You may have found evidence to the contrary that motivated its inclusion in its current state and I'd be happy to hear it.

A number of other potential small improvements.

File: .coveralls.yml
Comment (aonomike) [reply .coveralls.yml]: Thanks @SherSpock I was following [this blog](https://coveralls.io/github/AgileVentures/sfn-client) to get this setup and there is a suggestion its needed if we are not using travis pro? Please see below
Screen Shot 2019-05-10 at 11 29 46 AM
Maybe we can pass this from a .env though I have seen it added to other repos like the wso-fe so maybe its not that sensitive.

File: jest.config.js
Comment (aonomike) [reply jest.config.js]: The former tells jest where to collect coverage from(we do not want it collecting coverate for cypress tests) while the latter tells jest to ignore running tests in files within the cypress directory as they should be run using cypress command.

File: package.json
Comment (aonomike) [reply package.json:31]: Thanks @SherSpock I just added a coma on this line. `test.debug` was added on an earlier PR so as to help with running the debugger for using breakpoints. But maybe something we may need to review the naming.

File: package.json
Comment (aonomike) [reply package.json]: The configuration for the jest config, wasnt showing the coverage when I tested locally. maybe its a problem with create-react-app. I added this script as I would get coverage because of that. Maybe I remove the configuration with jest config.

File: jest.config.js
Comment (aonomike) [reply jest.config.js]: This was actually copy paste job. I will investigate

File: package.json
Comment (aonomike) [reply package.json]: makes sense. Will make updates

File: jest.config.js
Comment (aonomike) [reply jest.config.js]: Yeah, investigating the right way to do the change this. Thanks

File: .coveralls.yml
Comment (yakryder) [reply .coveralls.yml]: The link to blog post you posted goes [here](https://coveralls.io/github/AgileVentures/sfn-client) so I wasn't able to see what information you were using for this work.

I read the image. That would seem to indicate a need. However, it conflicts with other reports I've seen. It also doesn't give a reason that established the need for the add, which should always raise an eyebrow. If you investigate further and determine this is essential, that's obviously the closing argument. But it should be clearly motivated, not something we're blindly following

Some relevant links:

https://stackoverflow.com/questions/39501417/how-can-i-connect-coveralls-and-travis-in-github
https://github.com/tapjs/node-tap/issues/210

Personally my feeling is the token should be removed from source, then rotated. Its presence on wso-fe is likely a mistake. In general any creds-like object in source should be looked at with suspicion. 

Still open to counterarguments on this specific instance.

File: jest.config.js
Comment (yakryder) [reply jest.config.js]: That's clear from the code. What I mean to ask is, is there a good reason to explicitly say "don't generate coverage" for tests that are never being run through jest?

File: package.json
Comment (yakryder) [reply package.json:31]: The `test:debug` script is outside the scope of anything but merely observational and supporting comments for my review. It appears I put the comment on the wrong line -- this comment should be for the new coveralls scripts. 

I stand by my opening statement on this review -- the one critical thing I would block merge for is the `.coveralls.yml`. 

_EDIT: And now the text coverage reporter issue_

My opinion on this issue is that the coverage scripts don't deserve to be first class citizens in the top-level scripts namespace, nor do they deserve to be excluded from the `test` namespace. The `test` namespace is actually the most logical currently existent namespace for it to live in.

File: package.json
Comment (yakryder) [reply package.json]: If that was the behavior with the current state of the jest config, it seems to me that the coverage is being generated but not shown. If that is the case, personally I might prefer to see this particular script named `test:show_coverage`, or perhaps the coverage-defaulting removed from the jest config and made explicit everywhere it is wanted.

If in fact the current jest config generates but does not display coverage reports in the test run, that would potentially point to some mistakes made by the jest team. Given their API, `collectCoverage` should enable both generation and display.

OK, so digging in a little further, you shouldn't have removed the 'text' `coverageReporter` in the config, and really you [don't actually need to specify reporters at all](https://jestjs.io/docs/en/configuration#coveragereporters-array-string), as they'll default to `["json", "lcov", "text", "clover"]`. 

I'm not aware of any liability to having clover there needlessly, though for build parsimony's sake I wouldn't object to seeing it fall off. Nonetheless the cost to us is so small and you don't know what tools your devs will be wanting to use. 

Personally I think I would axe the clover reporter and put it back in the event a developer on the team asked for it.

I wasn't sure what was motivating the reporter-axing decision but I didn't ask about it on our first pass through. The text reporter needs to be restored.

**Comment by yakryder** [comment]
Just to update on this PR: 

**Essentials for which I will block merge**
1) Clearly motivate or change the coveralls repo token access method. If changed, rotate the token if this is reasonably easy to do
2) Restore the text coverage reporter to the jest config

**Good**
1) Replace deprecated method with current equivalent

**As you see fit**
All other changes, including namespacing and potentially unnecessary jest config parameters and build script

**Review by yakryder — CHANGES_REQUESTED** [review]
Please
1) Clearly motivate or change the coveralls repo token access method. If changed (i.e removed from source), rotate the token if this is reasonably easy to do
2) Restore the text coverage reporter to the jest config

File: package.json
Comment (aonomike) [reply package.json:31]: @SherSpock we have discussed this in the #weekend-collab and decided to remove the jest.config.js file since the test coverage is currently generated by yarn test --coverage and not with the jest.config.js set up. We will create a ticket to configure the jest configuration.

File: package.json
Comment (aonomike) [reply package.json]: I have updated the ticket to have `yarn test:coverage` to generate the coverage report.

File: .coveralls.yml
Comment (aonomike) [reply .coveralls.yml]: Thank you @SherSpock I have deleted the file `.coveralls.yml` as its not needed. See [this](https://github.com/lemurheavy/coveralls-public/issues/208) as the motivating factor.

**Review by yakryder — APPROVED** [review]
Looks good once the merge conflict is fixed. I like the decision to remove the jest config from this work -- constrains and simplifies.

Going ahead and approving since I know it won't go in without the merge conflict fix and that that'll be handled appropriately.

**Comment by yakryder** [comment]
@aonomike 
Sorry I missed your additional review request. Please feel free to follow up with me in the future on GitHub and/or Slack if I don't respond in a timely fashion to a PR that is in review.

Unfortunately, I can't look at this any further for a couple weeks. I'm sequestered away and really shouldn't even be posting this.

Hopefully the rubber stamp approval will get this moving again

**Comment by aonomike** [comment]
Thanks @SherSpock conflicts resolved

**Pull request merged**
